derive_client/_bridge/client.py

```diff
@@ -165,45 +166,89 @@ def _load_withdraw_wrapper(self) -> Contract:
+            remote_abi = json.loads(remote_abi_path.read_text())
+            src_abi, tgt_abi = (remote_abi, derive_abi) if is_deposit else (derive_abi, remote_abi)
+            src = get_contract(src_w3, src_addr, abi=src_abi)
+            tgt = get_contract(tgt_w3, tgt_addr, abi=tgt_abi)
+            src_event, tgt_event = src.events.OFTSent(), tgt.events.OFTReceived()
+            return BridgeContext(src_w3, tgt_w3, src, src_event, tgt_event)
+
+        elif bridge_type == BridgeType.SOCKET and currency is not Currency.DRV:
+            erc20_abi = json.loads(ERC20_ABI_PATH.read_text())
+            socket_abi = json.loads(SOCKET_ABI_PATH.read_text())
+
+            if is_deposit:
+                token_data: NonMintableTokenData = self.derive_addresses.chains[self.remote_chain_id][currency]
+                token_contract = get_contract(src_w3, token_data.NonMintableToken, abi=erc20_abi)
+            else:
+                token_data: MintableTokenData = self.derive_addresses.chains[ChainID.DERIVE][currency]
+                token_contract = get_contract(src_w3, token_data.MintableToken, abi=erc20_abi)
+
+            src_addr = SocketAddress[src_chain.name].value
+            tgt_addr = SocketAddress[tgt_chain.name].value
+            src_socket = get_contract(src_w3, address=src_addr, abi=socket_abi)
+            tgt_socket = get_contract(tgt_w3, address=tgt_addr, abi=socket_abi)
+            src_event, tgt_event = src_socket.events.MessageOutbound(), tgt_socket.events.ExecutionSuccess()
+            return BridgeContext(src_w3, tgt_w3, token_contract, src_event, tgt_event)
+
+        raise ValueError(f"Unsupported bridge_type={bridge_type} for currency={currency}.")
+
+    def _resolve_socket_route(
+        self,
+        direction: Literal["deposit", "withdraw"],
+        currency: Currency,
+    ) -> tuple[MintableTokenData | NonMintableTokenData, Address]:
+        is_deposit = direction == "deposit"
+        src_chain, tgt_chain = (
+            (self.remote_chain_id, ChainID.DERIVE) if is_deposit else (ChainID.DERIVE, self.remote_chain_id)
+        )
+
+        if (src_token_data := self.derive_addresses.chains[src_chain].get(currency)) is None:
+            msg = f"No bridge path for {currency.name} from {src_chain.name} to {tgt_chain.name}."
+            raise BridgeRouteError(msg)
+        if (tgt_token_data := self.derive_addresses.chains[tgt_chain].get(currency)) is None:
+            msg = f"No bridge path for {currency.name} from {tgt_chain.name} to {src_chain.name}."
+            raise BridgeRouteError(msg)
+
+        if tgt_chain not in src_token_data.connectors:
+            msg = f"Target chain {tgt_chain.name} not found in {src_chain.name} connectors."
+            raise BridgeRouteError(msg)
+        if src_chain not in tgt_token_data.connectors:
+            msg = f"Target chain {src_chain.name} not found in {tgt_chain.name} connectors."
+            raise BridgeRouteError(msg)
```

Comment:
we verify BOTH SIDES of the bridge exist before we continue. Technically only 1 is these (first of each set of two conditions) is necessary to bridge one way